lib/data/multipleBackendGateway.js

```diff
@@ -77,14 +173,28 @@ const multipleBackendGateway = {
                     }
                     cb();
                 });
+            } else if (client.clientType === 'aws_s3') {
+                if (deep) {
```

Comment:
is deep a boolean?  what determines its value/what is calling it? why is this only needed for the aws backend?